services/weatherAPI/weather_service.py

```diff
@@ -1,40 +1,82 @@
 import requests
+from abc import ABC, abstractmethod
+import urllib
+
 from services.ApiError import ApiError
 from util import Singleton
-from abc import ABC, abstractmethod
 from services.preferences import PrefService
 
 
+
 class WeatherAdapterModule(ABC):
```

Comment:
@timozuther I had struggled to get the weather for my home town which appears to have a weird name in the Geocoding Service, so I checked the Weather API and saw that you can also provide coordinates (lat, lon) instead of city names. I modified the `update` method accordingly. 

Doing that I saw that `get_weather` could be more fit than `update_weather` because only get information without modifying state on the remote.